app/src/main/java/com/edulectronics/tinycircuit/Models/Components/Component.java

```diff
@@ -17,34 +18,31 @@
  * This is the component base class. All components have to be derived from this base.
  */
 public abstract class Component implements IComponent {
-    protected List<Input> inputs = new ArrayList<Input>();
-    protected List<Output> outputs = new ArrayList<Output>();
 
-       protected double outputVoltage;
+    private double resistance;
+
+    protected List<ConnectionPoint> connectionPoints = new ArrayList<ConnectionPoint>(4);
 
     public Component() {
-        this.inputs.add(new Input(this));
-        this.outputs.add(new Output());
+        this.connectionPoints.add(new ConnectionPoint(this, ConnectionPointOrientation.Left));
+        this.connectionPoints.add(new ConnectionPoint(this, ConnectionPointOrientation.Right));
     }
 
-    public boolean hasOutputConnection() {
-        for (Output o : outputs) {
-            if (o.hasOutputConnection())
+    public boolean hasOutputConnection(ConnectionPoint connectionpoint) {
```

Comment:
Ik zou persoonlijk nog een aparte methode maken getOutgoingConnections(). hasOutputConnection() moet naar mijn mening alleen doen wat hij zegt, namelijk bepalen of er outputs zijn en zich verder niet bemoeien met de lijst van connections. Dus dan iets als `for (ConnectionPoint c : getOutgoingConnections(connectionPoint)) { ... }`